lacework/resource_lacework_alert_profile_test.go

```diff
@@ -0,0 +1,26 @@
+package lacework
+
+import (
+	"testing"
+
+	"github.com/lacework/go-sdk/api"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAlertProfileTemplateToArrayOfType(t *testing.T) {
+	var (
+		input        = []map[string]string{{"name": "alertName", "event_name": "alertEventName", "description": "alertDescription", "subject": "alertSubject"}}
+		d            = resourceLaceworkAlertProfile()
+		testResource = d.TestResourceData()
+		alerts       []api.AlertTemplate
+		actual       = []api.AlertTemplate{{Name: "alertName", EventName: "alertEventName", Description: "alertDescription", Subject: "alertSubject"}}
+	)
+
+	_ = testResource.Set("alert", input)
+
+	err := castSchemaSetToArrayOfAlertTemplate(testResource, "alert", &alerts)
+	assert.NoError(t, err)
+	assert.Equal(t, alerts, actual,
+		"%s did not match expected value: %s", actual, alerts,
+	)
```

Comment:
this test is really small, I feel we are not covering all cases with only one test like this, we should use a table test with multiple cases

Reply:
The cast is now specific to alert profiles rather than generic, going to leaving the test as is.